tests/run-e2e.sh

```diff
@@ -239,6 +240,39 @@ run_e2e() {
 declare -i ARGS_PARSED=0
 
 parse_args() {
```

Comment:
lets use the `cmd.sub-commmand`  trick to add sub-commands.  `main` should check for `cmd.$1` function and execute it.